packages/riverpod/lib/src/framework/base_provider.dart

```diff
@@ -1041,6 +998,12 @@ abstract class ProviderStateBase<Created, Listened> {
   @protected
   void valueChanged({Created? previous});
 
+  /// Updates the currently exposed value.
+  ///
+  /// This will usually be called by [ProviderReference.setState].
+  @protected
+  void exposedValueChanged(Listened newValue);
```

Comment:
The default behaviour should be to mutate `exposedValue`. There's no need to implement this method in all providers